docs/extensions/controllerregistration.md

```diff
@@ -199,7 +199,8 @@ The `.spec.deployment` resource allows to configure a deployment `policy`.
 There are the following policies:
 
 * `OnDemand` (default): Gardener will demand the deployment and deletion of the extension controller to/from seed clusters dynamically. It will automatically determine (based on other resources like `Shoot`s) whether it is required and decide accordingly.
-* `Always`: Gardener will demand the deployment of the extension controller to seed clusters independent of whether it is actually required or not. This might be helpful if you want to add a new component/controller to all seed clusters by default. Another use-case is to minimize the durations until extension controllers are deployed and ready in case you have highly fluctuating seed clusters.
+* `Always`: Gardener will demand the deployment of the extension controller to seed clusters independent of whether it is actually required or not. This might be helpful if you want to add a new component/controller to all seed clusters by default. Another use-case is to minimize the durations until extension controllers get deployed and ready in case you have highly fluctuating seed clusters.
+* `AlwaysIfShoots`: Similar to `Always`, but if the seed does not have any shoots then the extension is not being deployed.
```

Comment:
I think `AlwaysIfShoots` reads a bit bumpy. What about `AlwaysWithShoots` or `AlwaysForShoots` or something similar. 
No strong opinion here.

Reply:
Agreed, I didn't like it too much but didn't have a better idea. We can go with `AlwaysWithShoots`, open for any other suggestions as well.

Reply:
Another idea would be just `WithShoots`. But I think, I would still like `AlwaysWithShoots` more.

@gardener/gardener-maintainers any better ideas / objections?

Reply:
I am fine with `AlwaysWithShoots`. WDYT about `AlwaysExceptSeedWithoutShoots`?

Reply:
Throwing `AlwaysExceptNoShoots` in :)

Reply:
+1 for `AlwaysExceptNoShoots`.

Reply:
+1, `AlwaysExceptNoShoots` sounds good.